content/docs/tech-insights/data-sources/index.md

```diff
@@ -0,0 +1,71 @@
+
+To use built-in Data Sources, you don’t need to do any set up. Jump to the next section to learn more about Checks.
+
+### Defining your own Data Source
+
+Roadie Tech Insights lets you define arbitrary Data Sources. Typically, custom Data Sources rely on calling an API or reading a configuration file in a repository. When adding a Data Source, you’ll also specify which Facts are extracted from said source.
+
+To add a Data Source, click on the “Add Data Source” button in the Data Sources page.
+
+![Add Data Source](./data-sources-overview.png)
+
+To set up a Data Source, you’ll be prompted to define a Data Provider and define which fields will be extracted as Facts from the data. Let’s start overviewing the options available for setting up a Data Provider:
+
+![Add Data Provider](./data-provider-step.png)
+
+1 - You must specify a type for that new Data Source. The _HTTP_ type lets you connect to an external API to pull in data, while the _Component repository file_ type lets you extract data from a file path in the corresponding repository of a given Component in your Catalog. As an example, let’s use “_Component repository file”_.
+
+2 - Set the path to the file you want to extract data from in repositories, starting from the root. This can be anything from JSON files to YAML files.
+
+3 - Try out what would be the response when testing specific entity from the location you have provided. If you were to get the `package.json` from a `sample-service` component, the Data Source would get something like this:
+
+![Dry run](./dry-run-result.png)
+
+Now that you have data, let’s define what Facts interest you. You’ll do this through the Fields Extraction section.
+
+![Fields extraction](./field-extraction.png)
+
+4 - Choose a parser to extract a Fact from the data obtained before. For the type “Component repository file” this can be either JSON or Regex parser type, while for “HTTP” data provider type, only JSON is supported. Keep in mind, JSON type may be used only for JSON type of files.
+
+JSON type of parser uses [JSONPath Plus](https://jsonpath-plus.github.io/JSONPath/docs/ts/) syntax to extract data from JSON, while Regex type uses [ECMAScript syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) to extract data from text.
+
+5 - If you’re using the JSON parser, specify a path from the root of the object. For example _“version”,_ or “_scripts.test”_. If you’re using the Regex parser, specify a valid expression with a capture group if extracting values. Please note the Regex does not need slashes at the start or end.
+
+Let's look how we would do it with an example. If we want to retrieve specific line from the following result:
+
+![Regex result](./dry-run-yaml-result.png)
+
+If we wanted to retrieve Node version we could write the following Regex:
+
+![Field extraction Regex](./field-extraction-Regex.png)
```

Comment:
This image is not working.

![Screenshot 2022-11-16 at 13 46 56](https://user-images.githubusercontent.com/562403/202197272-c9cae311-9f54-4c86-b7ae-3c72ca91c243.png)